### DISCUSSION TITLE
Using the login page session

### DISCUSSION THREAD

**Discussion by davebarda** [ory/hydra#2866] [Q&A] [answered]
As state parameter is being used to verify that the request was originated from the relying party by the user agent, why the CSRF cookie is needed?

**Comment by vinckr** [comment, marked as answer]
Hello @davebarda 
Apologies for the late answer.

This goes beyond my understanding of our CSRF prevention mechanisms and I assume you are already familiar with [CSRF](https://cheatsheetseries.owasp.org/cheatsheets/Cross-Site_Request_Forgery_Prevention_Cheat_Sheet.html) in general? 

Was your question related to a specific issue or just out of interest? It would require some digging into the internals... 
(also [this](https://www.ory.sh/kratos/docs/debug/csrf/) might be relevant)

**Reply by vinckr** [reply]
Hey @davebarda ,
Just checking in if your question was answered or if there was still something unclear.
I will mark this as answered for now, but feel free to come back to the discussion any time.

Thanks a ton! 🐝
